src/casanovoutils/mgfutils.py

```
@@ -37,30 +37,23 @@ def shuffle(self) -> None:
         self.spectra = spectra_iter
 
-    def downsample(self, k: int = 1) -> None:
+    def spectra_per_peptide(self, k: int = 1) -> None:
         """
-        Downsample spectra by peptide sequence.
+        Retain up to k spectra per peptide using reservoir sampling.
 
-        Delegates to :func:`downsample_spectra`, which groups spectra by peptide
-        sequence (typically ``spectrum["params"]["seq"]``) and retains up to
-        ``k`` spectra per unique peptide.
+        Makes a single streaming pass, so memory usage is O(unique peptides × k)
```

Ruff (0.15.9)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)